geth/node/node.go

```diff
@@ -162,13 +163,13 @@ func activateShhService(stack *node.Node, config *params.NodeConfig) error {
 		logger.Info("SHH protocol is disabled")
 		return nil
 	}
-
-	serviceConstructor := func(*node.ServiceContext) (node.Service, error) {
+	var whisperService *whisper.Whisper
```

Comment:
We can initialize `whisperServiceConfig` and `whisperService` here instead in a callback passed to `stack.Register()`.

Reply:
i think it is better to keep initializing them when node is started

Reply:
Oh right. Forgot about the fact that `activateShhService` is called during node creation, not start. 

One more thing, it's possible to retrieve `Whisper` from `node.ServiceContext` so that registering both services can independent from the code perspective (ofc registering `shhext` would need to throw an error if Whisper is not available).